app/services/rad_sendgrid_status_receiver.rb

```diff
@@ -64,6 +65,16 @@ def process_status!
       end
     end
 
+    def update_contact_log!
+      return unless suppression?
+
+      contact_log.contact_log_recipients.find_by!(email: email).update! email_status: event
```

Comment:
this will hard fail if 1) the contact log can't be found and 2) the contact log recipient can't be found with the email address of the suppression event from SendGrid - thinking we let it roll and see how it goes once in the wild - in theory it should never fail

Reply:
should we  check to see if there are multiple recipients with the same email address and update them all, or perhaps prevent email from being duplicated within a contact log's contact log recipients?

Reply:
I added a raise so it will blow up if it happens and we can investigate, also added a related spec